### PR TITLE
apply spec updates to c++ server implementation

### PR DESCRIPTION
**Pull request by achim-k**
### Public-Facing Changes
Apply changes wrt. service definition to example server implementations

### Description

- [X] C++
- [X] Python
- [X] ~Typescript~ (done in #579)

### CODE REVIEW

File: cpp/foxglove-websocket/src/serialization.cpp [hunk 1]
```diff
@@ -103,17 +103,63 @@ void to_json(nlohmann::json& j, const Service& service) {
     {"id", service.id},
     {"name", service.name},
     {"type", service.type},
-    {"requestSchema", service.requestSchema},
-    {"responseSchema", service.responseSchema},
   };
+
+  if (service.request.has_value()) {
+    j["request"] = service.request.value();
+  }
```
Comment (jtbandes) [inline cpp/foxglove-websocket/src/serialization.cpp:110]: nit: possible cleanup
```suggestion
  if (service.request) {
    j["request"] = *service.request;
  }
```

File: cpp/foxglove-websocket/src/serialization.cpp [hunk 2]
```diff
@@ -103,17 +103,63 @@ void to_json(nlohmann::json& j, const Service& service) {
...
   p.name = j["name"].get<std::string>();
   p.type = j["type"].get<std::string>();
-  p.requestSchema = j["requestSchema"].get<std::string>();
-  p.responseSchema = j["responseSchema"].get<std::string>();
+
+  {
+    const auto it = j.find("request");
+    p.request = it == j.end() ? std::optional<ServiceRequestDefinition>(std::nullopt)
+                              : std::make_optional(it->get<ServiceRequestDefinition>());
+  }
```
Comment (jtbandes) [inline cpp/foxglove-websocket/src/serialization.cpp:131]: nit: possible cleanup
```suggestion
  if (const auto it = j.find("request"); it != j.end()) {
    p.request = it->get<ServiceRequestDefinition>();
  }
```

File: python/src/foxglove_websocket/types.py [hunk 3]
```diff
@@ -142,11 +142,24 @@ class Channel(ChannelWithoutId):
...
+
+
 class ServiceWithoutId(TypedDict):
     name: str
     type: str
-    requestSchema: str
-    responseSchema: str
+    request: Optional[ServiceRequestDefinition]
+    response: Optional[ServiceResponseDefinition]
+    requestSchema: Optional[str]
```
Comment (jtbandes) [inline python/src/foxglove_websocket/types.py:161]: nit: could add comments or docstrings here to mirror your deprecation comments in the C++ one

**Review by jtbandes — APPROVED** [review]

File: cpp/foxglove-websocket/src/serialization.cpp (hunk 2 above)
Comment (achim-k) [reply cpp/foxglove-websocket/src/serialization.cpp:131]: much better, thanks!

**Pull request merged**